spec-tests/src/test/java/org/candlepin/spec/environments/EnvironmentSpecTest.java

```diff
@@ -756,6 +762,378 @@ public void shouldRegenerateOnlyEntitlementsAffectedByDeletedEnvironment() {
+        OwnerDTO owner = admin.owners().createOwner(Owners.random());
+
+        EnvironmentDTO environment = admin.owners()
+            .createEnvironment(owner.getKey(), Environments.random());
+
+        EnvironmentDTO update = new EnvironmentDTO()
+            .description("long".repeat(256));
+
+        assertBadRequest(() -> admin.environments()
+            .updateEnvironment(environment.getId(), update));
+
+        EnvironmentDTO result = admin.environments()
+            .getEnvironment(environment.getId());
+
+        assertEquals(environment.getDescription(), result.getDescription());
+    }
+
+    @Test
+    public void shouldAllowUpdatingContentPrefixWithEnvironmentUpdate() {
+        OwnerDTO owner = admin.owners().createOwner(Owners.random());
+
+        EnvironmentDTO environment = admin.owners()
+            .createEnvironment(owner.getKey(), Environments.random());
+
+        EnvironmentDTO update = new EnvironmentDTO()
+            .contentPrefix("updated contentprefix");
+
+        EnvironmentDTO output = admin.environments()
+            .updateEnvironment(environment.getId(), update);
+
+        assertEquals(update.getContentPrefix(), output.getContentPrefix());
+
+        EnvironmentDTO fetched = admin.environments()
+            .getEnvironment(environment.getId());
+
+        assertEquals(update.getContentPrefix(), fetched.getContentPrefix());
+    }
+
+    @Test
+    public void shouldAllowClearingContentPrefixWithEnvironmentUpdate() {
+        OwnerDTO owner = admin.owners().createOwner(Owners.random());
+
+        EnvironmentDTO environment = admin.owners()
+            .createEnvironment(owner.getKey(), Environments.random());
+
+        EnvironmentDTO update = new EnvironmentDTO()
+            .contentPrefix("");
+
+        EnvironmentDTO output = admin.environments()
+            .updateEnvironment(environment.getId(), update);
+
+        assertNull(output.getContentPrefix());
+
+        EnvironmentDTO fetched = admin.environments()
+            .getEnvironment(environment.getId());
+
+        assertNull(fetched.getContentPrefix());
+    }
+
+    @Test
+    public void shouldNotAllowLongContentPrefixWithEnvironmentUpdate() {
+        EnvironmentDTO environment = admin.owners()
+            .createEnvironment(this.owner.getKey(), Environments.random());
+
+        EnvironmentDTO update = new EnvironmentDTO()
+            .contentPrefix("long".repeat(256));
+
+        assertBadRequest(() -> admin.environments()
+            .updateEnvironment(environment.getId(), update));
+
+        EnvironmentDTO result = admin.environments()
+            .getEnvironment(environment.getId());
+
+        assertEquals(environment.getContentPrefix(), result.getContentPrefix());
+    }
+
+
+    /**
+     * These tests verify we don't update fields that either should not be updated (ID, owner) or
+     * need special juggling for cert and content access (env content)
+     */
```

Comment:
Do we need to add tests here validating that the `created` and `updated` fields should not be updated?

Reply:
fixed